### PR TITLE
Make the vsync startup warning hidable

### PR DESCRIPTION
**Pull request by Teteros**
Since 5b51eb3 the vsync warning is set to always show up during startup
if vsync is turned off.

This is problematic with Intel/AMD drivers on linux, when using TearFree
(on in radeon driver by default, not sure about Intel)

This commit adds a button for the user to be able to suppress this warning.

### CODE REVIEW

**Comment by Teteros** [comment]
As a side note, the radium window/menus flickered quite bad with vsync on.
I've tracked the problem down on my side though. What enlightenment calls "overrides" for its compositor wasn't fast enough on my machine by default it seems.

scrots

So if anyone else has slow drawing/flickering (on AMD at least)
Playing around with the compositor might fix it for you as well.

File: OpenGL/Widget.cpp [hunk 1]
```diff
@@ -1741,9 +1741,20 @@ QWidget *GL_create_widget(QWidget *parent){
...
+	  vector_t v = {};
+	  VECTOR_push_back(&v,"Ok");
+	  VECTOR_push_back(&v,"Don't show this message again");
+
+	  int result = GFX_Message(&v, "Warning: VSync disabled. You probably don't want to do that. You can enable vsync under\nEdit -> Preferences -> OpenGL -> Vertical Blank");
+
+	  if (result==1)
+		  SETTINGS_write_bool("show_vsync_warning_during_startup", false);
+
+	  GL_set_pause_rendering_on_off(true);
```
Comment (kmatheussen) [inline OpenGL/Widget.cpp]: Copy/paste error?

**Comment by kmatheussen** [comment]
> This is problematic with Intel/AMD drivers on linux, when using TearFree
> (on in radeon driver by default, not sure about Intel)

Why is vsync problematic when using intel/amd on linux + tearfree?

**Comment by Teteros** [comment]
>Copy/paste error?

Quite possibly. Please correct me "if" is wrong as I basically copied the same if statement present in your AMD/Intel warnings:
(is `GL_set_pause_rendering_on_off(true);` needed?)`

https://github.com/Teteros/radium/blob/db49158586113463e38f9cc8ad557030aafd234d/OpenGL/Widget.cpp#L1831-L1871
https://github.com/Teteros/radium/blob/db49158586113463e38f9cc8ad557030aafd234d/OpenGL/Widget.cpp#L1877-L1917
https://github.com/Teteros/radium/blob/db49158586113463e38f9cc8ad557030aafd234d/OpenGL/Widget.cpp#L1938-L2009

Though I did recompile radium with that commit and it works as expected, 
The `show_vsync_warning_during_startup` boolean is stored in the config and vsync warning doesn't show up if it's false.

>Why is vsync problematic when using intel/amd on linux + tearfree?

Just did more testing and it really isn't problematic as I thought. I figured vsync caused my studder/flickering issues at the time when I posted this pull request, but it was the compositor "not keeping up" ([see above](https://github.com/kmatheussen/radium/pull/934#issuecomment-295146047)) because of a heavy effect.

Having tearfree (or vsync turned on in compositor settings) does not seem to affect radium at all, unless I'm missing something.
Not having vsync turned on in radium gives choppy playback, as expected.

Still, perhaps the AMD/Intel warnings are old, and the behaviour is changed (as it seems from my tests)

You suggest turning off vsync if using TearFree here:
https://github.com/Teteros/radium/blob/db49158586113463e38f9cc8ad557030aafd234d/OpenGL/Widget.cpp#L1963-L1969
https://github.com/Teteros/radium/blob/db49158586113463e38f9cc8ad557030aafd234d/OpenGL/Widget.cpp#L1841
https://github.com/Teteros/radium/blob/db49158586113463e38f9cc8ad557030aafd234d/OpenGL/Widget.cpp#L1886

**Comment by kmatheussen** [comment]
> (is GL_set_pause_rendering_on_off(true); needed?)

That's the line I was commenting on. It makes no sense to call this function here. Looks like a copy/paste error. Othervice the pull request looks fine.

I'm a little bit reluctant about this though. I'm pretty sure vsync should only be enabled if using a misbehaving driver, such as amd or intel on linux. The reason for the warning, though, is because a couple of people have disabled vsync, and got really horrible performance.

**Comment by kmatheussen** [comment]
It could also seem like a good idea to disable vsync since that's a way to increase performance in some computer games. But this is definitely not the case for Radium, at least normally.

**Comment by Teteros** [comment]
Added explanation why disabling vsync might be a bad idea, but left the option to suppress the warning for next startup if the user really wants it.

Looks good?

**Comment by kmatheussen** [comment]
Thanks. I think I'll change it later though, so that the "don't show again" option is only showed if using intel or amd on linux.

**Pull request merged**